### PR TITLE
Implement checks for vertex attribs enabled as arrays without a bound buffer

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21072)

### CODE REVIEW

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -3766,3 +3775,53 @@ impl UniformSetterType {
...
+        *self.elements.borrow_mut() = other.elements.borrow().clone();
+    }
+
+    pub fn borrow(&self) -> Ref<FnvHashMap<u32, (bool, Option<Dom<WebGLBuffer>>)>> {
+        self.elements.borrow()
+    }
+
+    fn remove_buffer(&self, buffer: &WebGLBuffer) {
+        self.elements.borrow_mut().retain(|_, v| {
+            if let Some(ref b) = v.1 {
```
Comment (emilio) [inline components/script/dom/webglrenderingcontext.rs]: nit: may be more compact with map_or.

**Review by emilio — APPROVED** [review]
Looks great, thanks!

File: components/script/dom/webglrenderingcontext.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/webglrenderingcontext.rs]: Fixed.

**Pull request merged**